CONTRIBUTING.rst

```diff
@@ -1,13 +1,105 @@
 Contributing to MLflow
 ======================
-We welcome community contributions to MLflow. This page describes how to develop/test your changes
-to MLflow locally.
-
-The majority of the MLflow codebase is in Python. This includes the CLI, Tracking Server,
+We welcome community contributions to MLflow. This page provides useful information about
+contributing to MLflow.
+
+.. contents:: **Table of Contents**
+  :local:
+  :depth: 3
+
+Contribution process
+####################
+The MLflow contribution process starts with filing a GitHub issue. MLflow defines four
+categories of issues: feature requests, bug reports, documentation fixes, and installation issues.
+Details about each issue type and the issue lifecycle are discussed in the `MLflow Issue Policy
+<https://github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md>`_.
+
+MLflow committers actively triage and respond to GitHub issues. In general, we recommend waiting
+for feebdack from an MLflow committer or community member before proceeding to implement a feature
+or patch. This is particularly important for
+`significant changes <https://github.com/mlflow/mlflow/blob/master/CONTRIBUTING.rst#write-designs-for-significant-changes>`_.
```

Comment:
I tried using RST references (`:ref:`) to link to sections within this document. Unfortunately, GitHub doesn't seem to render RST references properly (see https://github.com/isaacs/github/issues/892), so I gave this a shot instead.